core/common/assets/js/api/core/data/cache.js

```diff
@@ -0,0 +1,216 @@
+import ArgsObject from 'elementor-assets-js/modules/imports/args-object';
+import LocalStorage from './stroages/local-storage';
+
+export default class Cache extends ArgsObject {
+	/**
+	 * Function constructor().
+	 *
+	 * Create cache.
+	 *
+	 * @param {Data} manager
+	 */
+	constructor( manager ) {
+		super( {} );
+
+		this.manager = manager;
+
+		this.storage = new LocalStorage();
+	}
+
+	/**
+	 * Function validateRequestData().
+	 *
+	 * @param {RequestData} requestData
+	 */
+	validateRequestData( requestData ) {
+		this.requireArgument( 'component', requestData );
+		this.requireArgumentType( 'command', 'string', requestData );
+		this.requireArgumentType( 'endpoint', 'string', requestData );
+	}
+
+	/**
+	 * Function getAsync().
+	 *
+	 * Receive from cache. the difference between getAsync() and get() is that receive return it as promise...
+	 * to fake fetch mechanism.
+	 *
+	 * @param {RequestData} requestData
+	 *
+	 * @return {(Promise|boolean)}
+	 */
+	getAsync( requestData ) {
+		this.validateRequestData( requestData );
+
+		const data = this.get( requestData );
+
+		if ( null !== data ) {
+			// If data comes from cache, add 'cache = hit' to requestData.
+			requestData.cache = 'hit';
+
+			return new Promise( async ( resolve ) => {
+				resolve( data );
+			} );
+		}
+
+		// TODO: Check if possible, always return promise and reject it.
+		return false;
+	}
+
+	/**
+	 * Function load().
+	 *
+	 * Load data to cache.
+	 *
+	 * The difference between load() and update() is that load, will modify the data anyway...
+	 * when update() will only modify exist objects/values.
+	 *
+	 * @param {RequestData} requestData
+	 * @param {*} data
+	 */
+	load( requestData, data ) {
+		this.validateRequestData( requestData );
+
+		const componentName = requestData.component.getNamespace(),
+			nakedEndpoint = requestData.endpoint.replace( componentName + '/', '' ),
```

Comment:
Consider picking a different terminology

Reply:
It is the second approach we tried... do you have an idea?

Reply:
How about `pureEndPoint`?